crates/uv-toolchain/src/discovery.rs

```diff
@@ -602,11 +608,23 @@ pub fn find_toolchains<'a>(
                     Err(_) => true,
                     Ok((_source, interpreter)) => {
                         version.matches_interpreter(interpreter)
-                            && interpreter.implementation_name() == implementation.as_str()
+                            && interpreter
+                                .implementation_name()
+                                .eq_ignore_ascii_case(implementation.into())
                     }
                 })
                 .map(|result| result.map(Toolchain::from_tuple).map(ToolchainResult::Ok))
         }),
+        ToolchainRequest::Key(request) => Box::new({
+            debug!("Searching for {request} in {sources}");
+            python_interpreters(request.version(), request.implementation(), sources, cache)
+                .filter(move |result| result_satisfies_system_python(result, system))
+                .filter(|result| match result {
+                    Err(_) => true,
+                    Ok((_source, interpreter)) => request.satisfied_by_interpreter(interpreter),
+                })
```

Comment:
Maybe... `result.map_or(true, |(_, interpreter)| request.satisfied_by_interpreter(interpreter))`?

Reply:
This feels familiar :) will do

Reply:
This seems... less clear sometimes?

```diff
diff --git a/crates/uv-toolchain/src/discovery.rs b/crates/uv-toolchain/src/discovery.rs
index 8dbec984..d6bb224b 100644
--- a/crates/uv-toolchain/src/discovery.rs
+++ b/crates/uv-toolchain/src/discovery.rs
@@ -619,9 +619,10 @@ pub fn find_toolchains<'a>(
             debug!("Searching for {request} in {sources}");
             python_interpreters(request.version(), request.implementation(), sources, cache)
                 .filter(move |result| result_satisfies_system_python(result, system))
-                .filter(|result| match result {
-                    Err(_) => true,
-                    Ok((_source, interpreter)) => request.satisfied_by_interpreter(interpreter),
+                .filter(|result| {
+                    result.as_ref().map_or(true, |(_, interpreter)| {
+                        request.satisfied_by_interpreter(interpreter)
+                    })
                 })
                 .map(|result| result.map(Toolchain::from_tuple).map(ToolchainResult::Ok))
         }),
 ```
 
 Somehow it seems more verbose.

Reply:
I think I'd still go `map_or` but honestly it's dealer's choice. I'm not _too much_ of a combinator snob.

Reply:
I'd prefer `.map(..).unwrap_or(true)` to `map_or`.